src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -629,6 +635,71 @@ public VerboseAddressBalancesResult GetAddressIndexerState(string[] addresses)
             return result;
         }
 
+        public LastBalanceDecreaseTransactionModel GetLastBalanceDecreaseTransaction(string address)
+        {
+            if (address == null)
+                return null;
+
+            (bool isQueryable, string reason) = this.IsQueryable();
+
+            if (!isQueryable)
+                return null;
+
+            int lastBalanceHeight;
+
+            lock (this.lockObject)
+            {
+                AddressIndexerData indexData = this.addressIndexRepository.GetOrCreateAddress(address);
+
+                AddressBalanceChange lastBalanceUpdate = indexData.BalanceChanges.Where(a => !a.Deposited).OrderByDescending(b => b.BalanceChangedHeight).FirstOrDefault();
+
+                if (lastBalanceUpdate == null)
+                    return null;
+
+                lastBalanceHeight = lastBalanceUpdate.BalanceChangedHeight;
+            }
+
+            // Height 0 is used as a placeholder height for compacted address balance records, so ignore them if they are the only record.
+            if (lastBalanceHeight == 0)
+                return null;
+
+            ChainedHeader header = this.chainIndexer.GetHeader(lastBalanceHeight);
+
+            if (header == null)
+                return null;
+
+            Block block = this.consensusManager.GetBlockData(header.HashBlock).Block;
+
+            if (block == null)
+                return null;
+
+            // Get the UTXO snapshot as of one block lower than the last balance change, so that we are definitely able to look up the inputs of each transaction in the next block.
```

Comment:
```suggestion
            // Get the UTXO snapshot as of one block lower than the last balance change, so that we are definitely able to look up the inputs of each transaction in the subsequent block.
```

Bumping CI